app/domain/authentication/authn_jwt/authenticator.rb

```diff
@@ -115,10 +116,17 @@ def audit_role_id
         ::Audit::Event::Authn::RoleId.new(
           role: role,
           account: account,
-          username: jwt_identity
+          username: audited_username
         ).to_s
       end
 
+      def audited_username
```

Comment:
Please validate with PO and in our Code (in other flows i think we had the same issue in OIDC) what we writing to the audit when the user is not exists, i am not sure empty string is the answer

Reply:
Already discussed this with @hilagross and its okay for our authenticator. The username is the identity. No identity no username its seems legit

Reply:
There will be enough information in the message of the audit to understand this

Reply:
+ @sashaCher WDYtT we should write in audit when we do not have the user value ?